include/picongpu/particles/atomicPhysics/rateCalculation/BoundFreeFieldTransitionRates.hpp

```diff
@@ -120,5 +152,118 @@ namespace picongpu::particles::atomicPhysics::rateCalculation
             // 1/ sim.unit.time()
             return rateADK_AU * timeConversion;
         }
+
+    public:
+        /** field ionization ADK rate for a given electric field strength
+         *
+         * @tparam T_ChargeStateDataBox instantiated type of dataBox
+         * @tparam T_AtomicStateDataBox instantiated type of dataBox
+         * @tparam T_BoundFreeTransitionDataBox instantiated type of dataBox
+         *
+         * @param eFieldNorm E-field vector norm, in sim.units.eField()
+         * @param ionizationPotentialDepression, in eV
+         * @param transitionCollectionIndex index of transition in boundBoundTransitionDataBox
+         * @param atomicStateDataBox access to atomic state property data
+         * @param boundBoundTransitionDataBox access to bound-bound transition data
+         *
+         * @return unit: 1/picongpu::sim.unit.time()
+         */
+        template<
+            typename T_EFieldType,
+            typename T_ChargeStateDataBox,
+            typename T_AtomicStateDataBox,
+            typename T_BoundFreeTransitionDataBox>
+        HDINLINE static float_X rateADKFieldIonization(
+            T_EFieldType const eFieldNorm,
+            float_X const ionizationPotentialDepression,
+            uint32_t const transitionCollectionIndex,
+            T_ChargeStateDataBox const chargeStateDataBox,
+            T_AtomicStateDataBox const atomicStateDataBox,
+            T_BoundFreeTransitionDataBox const boundFreeTransitionDataBox)
+        {
+            if(eFieldNorm == 0._X)
+                return 0._X;
+
+            ResultStruct input = getInput(
```

Comment:
Have you given structured bindings a try? [This](https://github.com/NVIDIA/libcudacxx/issues/316) looks promising, although admittedly I didn't check if we support versions older than this (and if this is even about `nvcc` and not `nvc++`).

Reply:
at least one curretnly supported nvcc version is not happy about structured bindings. I get a `using __host function in __host__device__ function` warning, would skip that for now.